File: services/api-gateway/src/main/java/com/supportme/pay/apigateway/ApiGatewayApplication.java
package com.supportme.pay.apigateway;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * Jedyny punkt wejścia REST dla web/mobile w docelowej architekturze.
 * Bez własnej bazy danych, bez logiki biznesowej — tłumaczy REST na gRPC
 * do serwisów domenowych (core-svc, docelowo gateway-svc).
 */
@SpringBootApplication
public class ApiGatewayApplication {

    public static void main(String[] args) {
        SpringApplication.run(ApiGatewayApplication.class, args);
    }
}

File: services/api-gateway/src/main/kotlin/com/supportme/pay/apigateway/ApiGatewayApplication.kt
package com.supportme.pay.apigateway

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

/**
 * Jedyny punkt wejścia REST dla web/mobile w docelowej architekturze.
 * Bez własnej bazy danych, bez logiki biznesowej — tłumaczy REST na gRPC
 * do serwisów domenowych (core-svc, docelowo gateway-svc).
 */
@SpringBootApplication
open class ApiGatewayApplication

fun main(args: Array<String>) {
    runApplication<ApiGatewayApplication>(*args)
}
